students/headieh_godwin/lesson04/assignment/basic_operations.py

```diff
@@ -0,0 +1,122 @@
+    LOGGER.info("adding new customers to database")
+    new = [add_customer(*customer) for customer in customers]
+    return new
+
+def search_customer(customer_id):
+    """This function will return a dictionary object with firstname,
+     lastname, email address and phone number of a customer or an
+     empty dictionary object if no customer was found."""
+
+    LOGGER.info('searching for customer_id: %s', customer_id)
+
+    the_customer = Customer.get_or_none(Customer.customer_id == customer_id)
+
+    if the_customer:
+        LOGGER.info('building dictionary for: %s', customer_id)
+        customer = {}
+        customer['firstname'] = the_customer.firstname
+        customer['lastname'] = the_customer.lastname
+        customer['phone_number'] = the_customer.phone_number
+        customer['email_address'] = the_customer.email_address
+        LOGGER.info('Customer %s is found', customer_id)
+    else:
+        customer = {} # return empty dict object for non-existing customer.
+        LOGGER.info('Customer %s is not found', customer_id)
+
+    return customer
+
+def search_customers(customer_ids):
+    """Searches multiple customers in the database"""
+    LOGGER.info("searching for customers %s, within database", customer_ids)
+    return [search_customer(customer_id) for customer_id in customer_ids]
+
+def delete_customer(customer_id):
+    """This function will delete a customer from the sqlite3 database."""
+    try:
+        with DB.transaction():
+            LOGGER.info("deleting customer_id: %s", customer_id)
+            info = Customer.get(Customer.customer_id == customer_id)
+            info.delete_instance()
+            LOGGER.info("customer_id: %s has been deleted", customer_id)
+    except IndexError:
+        LOGGER.info('customer_id: %s does not exist - nothing has been deleted', customer_id)
+
+def delete_customers(customer_ids):
+    """Delete a list of customers from the database"""
+    LOGGER.info("deleting customer %s from database", customer_ids)
+    return [delete_customer(customer_id) for customer_id in customer_ids]
+
+def update_customer_credit(customer_id, credit_limit):
+    """This function will search an existing customer by customer_id
+    and update their credit limit or raise a ValueError exception if
+    the customer does not exist."""
+    try:
+        with DB.transaction():
+            customer = Customer.get(Customer.customer_id == customer_id)
+            customer.credit_limit = credit_limit
+            customer.save()
+            LOGGER.info("Datatbase updated successfully")
+    except IndexError:
+        LOGGER.info("customer_id does not exist")
+
+def list_active_customers():
+    """This function will return an integer with the number of customers
+    whose status is currently active."""
+    return len([customer.customer_id for customer in Customer.select().where(Customer.status)])
+    #active_count = Customer.select().where(Customer.status).count()
```

Comment:
Always avoid leaving commented code (it's ok to do for a couple of hours while testing something). As a rule, someone else will uncomment it and cause a disaster :-)